sds_data_manager/constructs/sdc_step_function.py

```diff
@@ -0,0 +1,116 @@
+from aws_cdk import Stack
+from aws_cdk import aws_s3 as s3
+from aws_cdk import aws_stepfunctions as sfn
+from aws_cdk import aws_stepfunctions_tasks as tasks
+from constructs import Construct
+
+from sds_data_manager.constructs.batch_compute_resources import FargateBatchResources
+from sds_data_manager.constructs.instrument_lambdas import InstrumentLambda
+
+
+class SdcStepFunction(Construct):
+    """Step Function Construct
+
+    Creates state machine using processing components.
+    """
+
+    def __init__(self,
+                 scope: Construct,
+                 construct_id: str,
+                 processing_step_name: str,
+                 processing_system: InstrumentLambda,
+                 batch_resources: FargateBatchResources,
+                 instrument_target: str,
+                 archive_bucket: s3.Bucket):
+        """SdcStepFunction Constructor
+
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        processing_step_name : str
+            The string identifier for the processing step
+        processing_system: BatchProcessingSystem
+            Batch processing system
+        batch_resources: FargateBatchResources
+            Fargate compute environment
+        instrument_target : str
+            Target data product
+        archive_bucket : str
+            S3 bucket
+        """
+        super().__init__(scope, construct_id)
+        print(archive_bucket.bucket_name)
+        # Reformat EventBridge Inputs
+        add_specifics_to_input = sfn.Pass(
+            self, "Reformat EventBridge Inputs",
+            parameters={
+                "TIMEOUT_TIME.$": "$.time",
+            }
```

Comment:
should we add/get information about file name or path that the processing step function should use? What do we use `TIMEOUT_TIME` for?

Reply:
You are right.  We will add onto this in the Trigger code I'll write. TIMEOUT_TIME is just the current time. I'll keep it as a placeholder, but it will likely change.